### ISSUE TITLE
Inconsistent warning using MultiHeadAttention with Masking 

### ISSUE THREAD

**Issue by fdtomasi** [keras-team/keras#20455]
Hello, when I try to use `MultiHeadAttention` (which supports masking), I get the following warning:
```
/opt/conda/envs/trdm/lib/python3.11/site-packages/keras/src/layers/layer.py:915: UserWarning: Layer 'query' (of type EinsumDense) was passed an input with a mask attached to it. However, this layer does not support masking and will therefore destroy the mask information. Downstream layers will not see the mask.
  warnings.warn(
/opt/conda/envs/trdm/lib/python3.11/site-packages/keras/src/layers/layer.py:915: UserWarning: Layer 'key' (of type EinsumDense) was passed an input with a mask attached to it. However, this layer does not support masking and will therefore destroy the mask information. Downstream layers will not see the mask.
  warnings.warn(
/opt/conda/envs/trdm/lib/python3.11/site-packages/keras/src/layers/layer.py:915: UserWarning: Layer 'value' (of type EinsumDense) was passed an input with a mask attached to it. However, this layer does not support masking and will therefore destroy the mask information. Downstream layers will not see the mask.
  warnings.warn(
```

However it seems that the mask is not really destroyed:

```python
from keras.layers import MultiHeadAttention, Masking
from keras import random
mha = MultiHeadAttention(num_heads=2, key_dim=3)
a = random.normal([1,3,4])
a[:, -1] = 0 
a = Masking()(a)
print(a._keras_mask)  # tensor([[ True,  True, False]], device='cuda:0')
b = mha(a, a)
print(b._keras_mask) # tensor([[ True,  True, False]], device='cuda:0')
```

[Einsum Layer](https://github.com/keras-team/keras/blob/master/keras/src/layers/core/einsum_dense.py) does not indeed specify `self.supports_masking = True` contrarily to MultiHeadAttention. I am not sure about the behaviour in general then. Is the mask preserved and warning can be ignored? Or are there instances which the mask will be destroyed? Because the [code](https://github.com/keras-team/keras/blob/master/keras/src/layers/layer.py#L931-L938) in `Layer` does not seem to preserve the mask when the warning is activated (so I am not sure how I am able to retain the mask in the example). Thank you!

**Comment by fchollet** [comment]
In this case the warning is safe to ignore, because the mask is tracked by the MHA layer. Let's silence it.

**Comment by fchollet** [comment]
This is now fixed.
